zilliqa/src/consensus.rs

```diff
@@ -1754,11 +1793,21 @@ impl Consensus {
             return Ok(TxAddResult::Duplicate(txn.hash));
         }
 
-        let account = self.state.get_account(txn.signer)?;
+        // Perform insertion under early state, if available
+        let early_account = match self.early_proposal.as_ref() {
```

Comment:
the code had a subtle error here, where new txns were being inserted under stale state/account info. this fix uses the state in the early proposal, if present.